### DISCUSSION TITLE
Why doesn't my OpenSSL build with macros.h?

### DISCUSSION THREAD

**Discussion by devlzcode** [openssl/openssl#25593] [Q&A] [answered]
I'm trying to build cURL with OpenSSL but it requires macros.h to exist and for some reason it doesn't exist in my OpenSSL build. Here is my script:
```sh
export ANDROID_NDK_HOME=$ANDROID_HOME/ndk/27.1.12297006
export HOST_TAG=darwin-x86_64
export TOOLCHAIN=$ANDROID_NDK_HOME/toolchains/llvm/prebuilt/$HOST_TAG
PATH=$TOOLCHAIN/bin:$PATH
../openssl/Configure android-arm -D__ANDROID_API__=31
make
cp ./libssl.a $TOOLCHAIN/sysroot/usr/lib/
cp ./libcrypto.a $TOOLCHAIN/sysroot/usr/lib/
cp -r ./include/openssl $TOOLCHAIN/sysroot/usr/include/
```

**Comment by mattcaswell** [comment, marked as answer]
The only files that will exist in your build directory are those files that are actually "built" in some way. For header files that typically means those generated from an input ".in" file, e.g. "asn1.h" is built from the "asn1.h.in" file in the source directory.

The "macros.h" header file is not built in anyway. So you will find it in the source include/openssl directory.

> cp -r ./include/openssl $TOOLCHAIN/sysroot/usr/include/

This is only copying the built openssl header files. Not those that exist already in the source include/openssl directory.

**Reply by levitte** [reply]
From the script above, it seems that all you need to add is this:

``` shell
cp ../openssl/include/openssl/*.h $TOOLCHAIN/sysroot/usr/include/
```

**Reply by devlzcode** [reply]
Thanks I didn't know that, I've never built a C library before 😓 . Just trying to build curl for android arm64-8va...
